spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/converters/PubSubMessageConverter.java

```diff
@@ -26,40 +30,58 @@
 import org.springframework.messaging.support.GenericMessage;
 
 /**
- * Generates {@link PubsubMessage} from {@link Message} and vice-versa.
+ * Generates {@link PubsubMessage} from {@link Message} with payload type of
+ * {@link String} and vice-versa. When converting to String, it defaults to using UTF-8
+ * @{link Charset}.
  *
- * @author João André Martins
+ * @author Ray Tsang
  */
-public class SimpleMessageConverter implements MessageConverter {
+public class PubSubMessageConverter implements MessageConverter {
+	private final Charset charset;
+
+	public PubSubMessageConverter() {
+		this(Charset.forName("UTF-8"));
+	}
+
+	public PubSubMessageConverter(Charset charset) {
+		this.charset = charset;
```

Comment:
Are we allowed for `null`s for this property ?

Reply:
it can't, should add assertion.